frappe/desk/page/leaderboard/leaderboard.js

```diff
@@ -0,0 +1,371 @@
+					data: {
+						datasets: [
+							{
+								values: graph_items.map(d => d.value)
+							}
+						],
+						labels: graph_items.map(d => d.name)
+					},
+					colors: ["light-green"],
+					format_tooltip_x: d => d[this.options.selected_filter_item],
+					type: "bar",
+					height: 140
+				};
+				new frappe.Chart(".leaderboard-graph", args);
+
+				notify(this, r);
+			}
+		});
+	}
+
+	get_leaderboard_data(me, res) {
+		if (res && res.message.length) {
+			me.message = null;
+			me.$container.find(".leaderboard-list").html(me.render_list_view(res.message));
+			me.setup_search($(me.parent).find('.list-item-container'));
+		} else {
+			me.$graph_area.hide();
+			me.message = __("No items found.");
+			me.$container.find(".leaderboard-list").html(me.render_list_view());
+		}
+	}
+
+	render_list_view(items = []) {
+
+		var html =
+			`${this.render_message()}
+			<div class="result" style="${this.message ? "display: none;" : ""}">
+				${this.render_result(items)}
+			</div>`;
+
+		return $(html);
+	}
+
+	render_result(items) {
+
+		var html =
+			`${this.render_list_header()}
+			${this.render_list_result(items)}`;
+		return html;
+	}
+
+	render_list_header() {
+		const _selected_filter = this.options.selected_filter
+			.map(i => frappe.model.unscrub(i));
+		const fields = ["rank", "name", this.options.selected_filter_item];
+		const filters = fields.map(filter => {
+			const col = frappe.model.unscrub(filter);
+			return (
+				`<div class="leaderboard-item list-item_content ellipsis text-muted list-item__content--flex-2
+					header-btn-base ${filter}
+					${(col && _selected_filter.indexOf(col) !== -1) ? "text-right" : ""}">
+					<span class="list-col-title ellipsis">
+						${col}
+					</span>
+				</div>`
+			);
+		}).join("");
+
+		const html =
+			`<div class="list-headers">
+				<div class="list-item list-item--head" data-list-renderer="List">${filters}</div>
+			</div>`;
+		return html;
+	}
+
+	render_list_result(items) {
+
+		let _html = items.map((item, index) => {
+			const $value = $(this.get_item_html(item, index+1));
+			const $item_container = $(`<div class="list-item-container">`).append($value);
+			return $item_container[0].outerHTML;
+		}).join("");
+
+		let html =
+			`<div class="result-list">
+				<div class="list-items">
+					${_html}
+				</div>
+			</div>`;
+
+		return html;
+	}
+
+	render_message() {
+
+		let html =
+			`<div class="no-result text-center" style="${this.message ? "" : "display: none;"}">
+				<div class="msg-box no-border">
+					<p>No Item found</p>
+				</div>
+			</div>`;
+
+		return html;
+	}
+
+	get_item_html(item, index) {
+		const company = this.options.selected_company;
+		const currency = frappe.get_doc(":Company", company).default_currency;
+		const fields = ["rank", "name", "value"];
+
+		const html =
+			`<div class="list-item">
+			${
+			fields.map(col => {
+				let val = col === "rank"? index: item[col];
+				let link = this.options.selected_doctype === "Energy Point Log"
```

Comment:
Leaderboard code should be doctype agnostic... can we move this to leaderboard config? May can we get fieldtype (basically df) with fields and based on that we can format.